### ISSUE TITLE
[Bug]: Cannot invoke `code-server` from integrated terminal

### ISSUE THREAD

**Issue by Emporea** [coder/code-server#5335]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Firefox 102
- Local OS: Win10
- Remote OS: Ubuntu
- Remote Architecture: 86
- `code-server --version`: 4.5.0 (4.4.0)

### Steps to Reproduce

1. docker run codeserver:latest
2. try to install extensions via code-server `code-server --install-extension ms-python`
3. asd

### Expected

Installation of Extension

### Actual

Console throws error
`/usr/lib/code-server/lib/vscode/bin/remote-cli/code-server: 12: /usr/lib/code-server/lib/vscode/node: not found`

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

**Comment by Emporea** [comment]
"Workaround" is a downgrade to 4.4.0
There it seems to work.

#5334 also has a bug appearently solved by downgrading.

**Comment by dx034** [comment]
I have the same issue but need 4.5.0 as it's required to run Github Copilot. Any other ideas for a workaround?

**Comment by benz0li** [comment]
> I _think_ what's going on is upstream `vscode` is injecting an update to the `PATH` environment variable to an undesirable location.

@jsjoeio I can confirm this.

JupyterLab Terminal:

```zsh
$ echo $PATH
/opt/code-server/bin:/opt/julia/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

code-server Terminal:

```zsh
$ echo $PATH
/opt/code-server/lib/vscode/bin/remote-cli:/opt/code-server/bin:/opt/julia/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

👉 Compare `/opt/code-server/bin/code-server` with `/opt/code-server/lib/vscode/bin/remote-cli/code-server`.

(With `/opt/code-server` being the root directory of my `code-server` installation)

**Comment by benz0li** [comment]
@jsjoeio Please add this issue to the [July 2022](https://github.com/coder/code-server/milestone/42) milestone.

**Comment by jsjoeio** [comment]
> The way code-server is released(/run?), it behaves like a remote installation. This might be the reason, why /opt/code-server/lib/vscode/bin/remote-cli is prepended to PATH.

Interesting 🤔 I'll chat with @code-asher and see how we can fix this. Thanks for bringing it up everyone!

**Comment by benz0li** [comment]
@code-asher Compare `/opt/code-server/bin/code-server` with `/opt/code-server/lib/vscode/bin/remote-cli/code-server`.

(`/opt/code-server` being the root directory of my `code-server` installation)

**Comment by fritterhoff** [comment]
So I might be wrong but the only location where upstream vs code prepends the PATH variable with the remote-cli folder is here:

https://github.com/microsoft/vscode/blob/619ab6d1ba7eef9ad418456a5fe5c7805f01a7ab/src/vs/server/node/extensionHostConnection.ts#L53-L59

**Comment by fritterhoff** [comment]
My pretty simple idea would be to check inside the code-server script it's location. 
So in case of being inside the `remote-cli` folder instead of the `bin` it maybe would be a solution to change the way the `node` location is detected.

**Comment by code-asher** [comment]
Or maybe we prevent it from getting added to the `PATH`.

**Comment by benz0li** [comment]
If

> `/opt/code-server/bin/code-server` is the right one.

you should

> patch to prevent it [`/opt/code-server/lib/vscode/bin/remote-cli`] from getting added to the PATH?

**Comment by fritterhoff** [comment]
I'm currently trying what happens if I simply remove the code-server script located in `remote-cli` folder.

**Comment by fritterhoff** [comment]
![grafik](https://user-images.githubusercontent.com/32478819/180129994-bed17e29-d94c-4eba-bea9-8c3026411458.png)
So removing the `code-server` script inside the `remote-cli` folder at least works in my case fine... Not sure if there are some negative side-effects I did not see in my first tests?

**Comment by benz0li** [comment]
@code-asher In code-server v4.4.0 there is `/opt/code-server/lib/vscode/bin/remote-cli/code-oss`.  
👉 `/opt/code-server/lib/vscode/bin/remote-cli` is prepended to `PATH`, too; but there is `code-oss` instead of `code-server`.

**Comment by benz0li** [comment]
In code-server v4.1.0 `/opt/code-server/vendor/modules/code-oss-dev/bin` is prepended to `PATH`; the directory only containing `helpers/{}`

**Comment by tjcran** [comment]
Will be resolved by #5360
